### DISCUSSION TITLE
Queries are not made again when navigating between routes without refreshing the page

### DISCUSSION THREAD

**Discussion by GuasaPlay** [Akryum/vue-apollo#1268] [Q&A]
Hello, I am new to graphql and I am trying to use vue-apollo to perform a query, I already managed to do my first query with smart query so far so good ... but I have a problem, I am making a query on the `/` path now If I change the route without refreshing the browser to  `/about`  and then return to the ` / ` route again, the query is not made again and I don't really know why it happens, it is as if there were `keep-alive` but not there are.

The only thing I could think of is that when the component is mounted to do a  `this.$Apollo.queries.workspaces.refetch()`  but I don't know if it's the best way and it doesn't affect performance.

Any suggestion or help would be very grateful.

This is my code where I am making the query:

```
<template>
  <div>
    <nuxt-link to="/about">About</nuxt-link>
    <div>
      <pre>
        {{ workspaces }}
      </pre>
    </div>
  </div>
</template>

<script>
import getWorkspaces from '@/graphql/queries/getWorkspaces.gql'
export default {
  data() {
    return {
      workspaces: {},
    }
  },
  apollo: {
    workspaces: {
      query: getWorkspaces,
    },
  },
  // mounted() {
  //   this.$apollo.queries.workspaces.refetch()
  // },
}
</script>
```

**Comment by Akryum** [comment]
https://www.apollographql.com/docs/react/caching/overview/
https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.watchQuery (`fetchPolicy`)
